pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternTopicsConsumerImpl.java

```diff
@@ -36,11 +50,119 @@ public PatternTopicsConsumerImpl(Pattern topicsPattern,
                               CompletableFuture<Consumer> subscribeFuture) {
         super(client, conf, listenerExecutor, subscribeFuture);
         this.topicsPattern = topicsPattern;
+
+        if (this.namespaceName == null) {
+            this.namespaceName = getNameSpaceFromPattern(topicsPattern);
+        }
+        checkArgument(getNameSpaceFromPattern(topicsPattern).toString().equals(this.namespaceName.toString()));
+
+        this.topicsChangeListener = new PatternTopicsChangedListener();
+        scheduledExecutor = (ScheduledExecutorService)listenerExecutor;
+        scheduledExecutor.scheduleAtFixedRate(() -> recheckTopics(), 60, 30, TimeUnit.SECONDS);
```

Comment:
The frequency of the check should probably be configurable, though it should have a minimum of, say 1min, to avoid "hitting refresh" too often